### ISSUE TITLE
"Ungültiger Maschinenbefehl"

### ISSUE THREAD

**Issue by TomCat112** [baycom/tfrec#6]
Hello,
i made the installation like you descripe in the wiki. After all i trie to test it. But with the parameter -D i just get this output:

Registering demod for TFA_1 KlimaLoggPro
Ung?ltiger Maschinenbefehl

What can be the fault?

Thanks for your reply and thank for the great work

Thomas

**Comment by baycom** [comment]
Sound like compiling for the wrong CPU, but maybe it's a real bug. Can you enable the -g debug-option in the makefile, "make clean; make" and run it again in gdb? Then the stack trace of the crash can optained by "thread apply all bt"

**Comment by rsneyer** [comment]
I had a similar error trying to get tfrec run on my Raspberry Pi Zero W.
Solution is to adjust the CXX flags in the Makefile:
..
CXXFLAGS=-O3 $(PROFILING) -mfpu=vfp -march=armv6 -mfloat-abi=hard -ftree-vectorize -ffast-math \
...

Details about compiler flags for different Raspberry models can be found in the raspbian FAQ or [here] (https://www.raspberrypi.org/forums/viewtopic.php?t=11629)

Have a great day,
Robert

**Comment by baycom** [comment]
There is now a adapted Makefile.raspi2, should also work for the Zero

**Comment by devZer0** [comment]
yes, it works. i'm using it on raspberry.  @TomCat112  , can you give a comment on the status so this ticket won't remain open forever ?
